bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQAbstractDeviceHandler.java

```diff
@@ -0,0 +1,820 @@
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        normalizeConfigurationsAndProperties();
+        lgPlatformType = String.valueOf(thing.getProperties().get(PROP_INFO_PLATFORM_TYPE));
+
+        Class<S> snapshotClass = getSnapshotClass();
+        try {
+            this.lastShot = snapshotClass.getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Snapshot class can't be instantiated. It most likely a bug", e);
+        }
+    }
+
+    private LGThinQBridgeHandler getAccountBridgeHandler() {
+        return Objects.requireNonNull(this.account, "BridgeHandler not initialized. It most likely a bug");
+    }
+
+    private void normalizeConfigurationsAndProperties() {
+        List.of(PROP_INFO_PLATFORM_TYPE, PROP_INFO_MODEL_URL_INFO, PROP_INFO_DEVICE_ID).forEach(p -> {
+            if (!thing.getProperties().containsKey(p)) {
+                thing.setProperty(p, (String) thing.getConfiguration().get(p));
+            }
+        });
+    }
+
+    protected static class AsyncCommandParams {
+        final String channelUID;
+        final Command command;
+
+        public AsyncCommandParams(String channelUUID, Command command) {
+            this.channelUID = channelUUID;
+            this.command = command;
+        }
+    }
+
+    /**
+     * Returns the simple channel UID name, i.e., without group.
+     *
+     * @param uid Full UID name
+     * @return simple channel UID name, i.e., without group.
+     */
+    protected String getSimpleChannelUID(String uid) {
+        String simpleChannelUID;
+        if (uid.indexOf("#") > 0) {
+            // I have to remove the channelGroup from de channelUID
+            simpleChannelUID = uid.split("#")[1];
+        } else {
+            simpleChannelUID = uid;
+        }
+        return simpleChannelUID;
+    }
+
+    /**
+     * Return empty string if null argument is passed
+     *
+     * @param value value to test
+     * @return empty string if null argument is passed
+     */
+    protected final String emptyIfNull(@Nullable String value) {
+        return Objects.requireNonNullElse(value, "");
+    }
+
+    /**
+     * Return the key informed if there is no correpondent value in map for that key.
+     *
+     * @param map map with key/value
+     * @param key key to search for a value into map
+     * @return return value related to that key in the map, or the own key if there is no correspondent.
+     */
+    protected final String keyIfValueNotFound(Map<String, String> map, String key) {
+        return Objects.requireNonNullElse(map.get(key), key);
+    }
+
+    @SuppressWarnings("null")
+    protected void startCommandExecutorQueueJob() {
+        if (commandExecutorQueueJob == null || commandExecutorQueueJob.isDone()) {
+            commandExecutorQueueJob = getExecutorService().submit(getQueuedCommandExecutor());
+        }
```

Comment:
Remove ALL null suppresions and handle the null inline. Otherwise it only hides potential problems instead of solving the cause.
In many cases it is as simple as assigning and using a local variable. For example:
```suggestion
    protected void startCommandExecutorQueueJob() {
        Future<?> commandExecutorQueueJob = this.commandExecutorQueueJob;
        if (commandExecutorQueueJob == null || commandExecutorQueueJob.isDone()) {
            this.commandExecutorQueueJob = getExecutorService().submit(getQueuedCommandExecutor());
        }
```

Reply:
This is one case that is impossible remove compilers NPE warning without supressing. Even knowing that is impossible to get NPE in this "if", the compiler complain about it:
![image](https://github.com/user-attachments/assets/18acd0eb-5dde-4281-b5d1-2e95be9eb9d0)

```
[WARNING] /home/nemer/repositories/Smart/smart-home/openhab-addons4/bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQAbstractDeviceHandler.java:[116,48] Potential null pointer access: The method getGenericSuperclass() may return null
[WARNING] /home/nemer/repositories/Smart/smart-home/openhab-addons4/bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQAbstractDeviceHandler.java:[197,48] Potential null pointer access: this expression has a '@Nullable' type

```
What should I do ?

Reply:
Done !